### DISCUSSION TITLE
Ordering, using a "magic number"

### DISCUSSION THREAD

**Discussion by brillout** [magne4000/universal-middleware#4] [Ideas]
`order?: 'pre' | 'post' | number`,

The order is determined after the magic number while `post` sets the number to `100`, `pre` sets it to `-100`, and the default is `0`.

It's far from perfect: to be able to resolve conflicts the magic number of other universal handlers may need to be changed. But it's the best solution I'm aware of for addressing the general problem of decentral ordering.

To alleviate issues a "range catalog recommendation" can be crafted over time. E.g. for authentication recommending `-50` to `-40`, for data fetching `30` to `40`, for SSR `50` to `60`, for error handling `-210` to `-200`. But we probably won't need this for a while as we'll control all universal handlers for the time being.

This will eventually be needed for [the Stem vision](https://github.com/magne4000/universal-handler/discussions/5).

#### Eject idea

When the user ejects the server code, then don't use the order number anymore: the order is then completely controlled by the user.

#### Only needed for the Stem vision

See: https://github.com/magne4000/universal-handler/discussions/2#discussioncomment-9595440.

**Comment by magne4000** [comment]
I had ChatGPT generate this enum for me: https://github.com/magne4000/universal-middleware/blob/509a6aad3ab63d256b888a2cf02add566f56116c/packages/core/src/const.ts#L22

Still not perfect, but a good starting point
